### PR TITLE
Fix documentation enforcement

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
We accidentally lost the way to set `PR Check` failure at some point.

### Example
The bad non-ok [status](https://api.github.com/repos/ClickHouse/ClickHouse/statuses/1bcf035bbdc693c5b16e30bf40caaf903ed15f33?per_page=1&page=118) was [overwritten](https://api.github.com/repos/ClickHouse/ClickHouse/statuses/1bcf035bbdc693c5b16e30bf40caaf903ed15f33?per_page=1&page=116) in 5 seconds after it was created.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [x]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit c22f1e566c0dcd9d044c8c4442e10ef5081372e1 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64890/c22f1e566c0dcd9d044c8c4442e10ef5081372e1/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>❌ failure</td></tr>
<tbody>
</table>

**Comment by Felixoid** [comment]
image

It works for the `pr-feature`

File: tests/ci/run_check.py [hunk 1]
```diff
@@ -146,51 +151,49 @@ def main():
...
-            f"expect adding docs for {Labels.PR_FEATURE}",
-            PR_CHECK,
-            pr_info,
-        )
-        # allow the workflow to continue
+        status = FAILURE
+        description = f"expect adding docs for {Labels.PR_FEATURE}"
+    # 3. But we allow the workflow to continue
 
+    # 4. And post only a single commit status on a failure
```
Comment (vdimir) [inline tests/ci/run_check.py:178]: On failure of another check, not documentation?
So, missing documentation will run workflow but will set mergable check to `failed` at the end, is it correct?

**Review by vdimir — APPROVED** [review]

File: tests/ci/run_check.py (hunk 1 above)
Comment (Felixoid) [reply tests/ci/run_check.py:178]: Right, `can_run` is set above in `should_run_ci_for_pr`.

**Pull request merged**